From 997fd38ec827eda4449f8716723bf772f0ff140d Mon Sep 17 00:00:00 2001
From: Andrii Anisov <andrii.anisov@gmail.com>
Date: Mon, 21 Nov 2016 19:01:55 +0200
Subject: [PATCH 4/4] salvator-x:dom0: Move all memory banks to a single node

Due to a weird "fdt memory fixup" in u-boot introduce a single memory
node.

Signed-off-by: Andrii Anisov <andrii.anisov@gmail.com>
---
 .../boot/dts/renesas/r8a7795-salvator-x-dom0.dts     | 20 ++++----------------
 1 file changed, 4 insertions(+), 16 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-dom0.dts b/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-dom0.dts
index 0f12b4a..d3fa193 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-dom0.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-dom0.dts
@@ -93,22 +93,10 @@
 	memory@48000000 {
 		device_type = "memory";
 		/* first 128MB is reserved for secure area. */
-		reg = <0x0 0x48000000 0x0 0x38000000>;
-	};
-
-	memory@500000000 {
-		device_type = "memory";
-		reg = <0x5 0x00000000 0x0 0x40000000>;
-	};
-
-	memory@600000000 {
-		device_type = "memory";
-		reg = <0x6 0x00000000 0x0 0x40000000>;
-	};
-
-	memory@700000000 {
-		device_type = "memory";
-		reg = <0x7 0x00000000 0x0 0x40000000>;
+		reg = <0x0 0x48000000 0x0 0x38000000>,
+		      <0x5 0x00000000 0x0 0x40000000>,
+		      <0x6 0x00000000 0x0 0x40000000>,
+		      <0x7 0x00000000 0x0 0x40000000>;
 	};
 
 	fixedregulator3v3: regulator@0 {
-- 
2.7.4

